go/flowctl/cmd-discover.go

```diff
@@ -259,15 +259,38 @@ func writeConfigStub(ctx context.Context, image string, w io.WriteCloser) error
 		} else if *node == nil {
 
 			var nn = new(yaml.Node)
-			nn.Style = yaml.FoldedStyle
+			nn.Style = yaml.FlowStyle // renders values inline with the keys instead of on the next line
+			nn.Kind = yaml.ScalarNode // seems required to get numbers and bools to render correctly
 
 			nn.FootComment =
 				fmt.Sprintf("%s\n%s", loc.Spec.Description, loc.Spec.Types)
 
 			if loc.Spec.MustExist {
 				nn.FootComment += " (required)"
 			}
```

Comment:
This is definitely outside the scope of this PR, but it looks like we could render the comments *above* each line, rather than below by using [`Node.HeadComment`](https://pkg.go.dev/gopkg.in/yaml.v3#Node) rather than `FootComment`. It's minor, but it was something that felt rather odd when I first saw it. If we're in here making these generated catalog files make a little more sense, it might be worth doing.

Reply:
Yeah, TBH that's my preference as well, but I can also see the appeal of putting it below for users who may not be using an editor with syntax highlighting. I also took an informal survey of one (@dyaffe ), and the response was that comments below felt easier to read. So ... :man_shrugging: